great_expectations/expectations/expectation.py

```diff
@@ -368,18 +368,12 @@ def __eq__(self, other: object) -> bool:
 
         # rendered_content is derived from the rest of the expectation, and can/should
         # be excluded from equality checks
-        exclude: set[str] = {"rendered_content"}
+        # notes and meta are simple metadata and should be excluded from equality checks
+        exclude: set[str] = {"rendered_content", "notes", "meta"}
```

Comment:
Should we really exclude these for equality? Why not just handle these kind of weird situations in the helper method above?